Dockerfile

```diff
@@ -14,11 +14,12 @@ ENV NODE_OPTIONS=--max_old_space_size=8000
 
 WORKDIR /tmp/grafana
 
-COPY package.json yarn.lock .yarnrc.yml ./
+COPY package.json project.json nx.json yarn.lock .yarnrc.yml ./
 COPY .yarn .yarn
 COPY packages packages
 COPY plugins-bundled plugins-bundled
 COPY public public
```

Comment:
do we actually need public at this stage? We should remove either this line or line 28 instead of copying it twice

Reply:
Good question. I've removed it from line 28 and it continues to build fine for me locally.

Reply:
As a side note, each one of these `COPY` lines adds another layer to the image. So consolidation to a single `COPY` for each block would optimize the image and reduce it's size a bit. Nothing we have to change, just something to keep in mind.

Reply:
yeah, they're separate because of the COPY semantics for folders iirc

Reply:
Looking into this further, yes I believe we need to copy them prior to `yarn install`. The decoupled plugins (which live inside public) are yarn workspaces and so have package.json files which the `yarn install` needs.

Reply:
fair enough, not much we can do in that case.